src/app/user/services/user-store.service.ts

```diff
@@ -1,19 +1,40 @@
-import { Injectable } from '@angular/core';
+import { Injectable } from "@angular/core";
+import { BehaviorSubject, Observable, tap } from "rxjs";
+import { UserService } from "./user.service";
 
 @Injectable({
-    providedIn: 'root'
+  providedIn: "root",
 })
 export class UserStoreService {
+  private name$$ = new BehaviorSubject<string | null>(null);
+  private isAdmin$$ = new BehaviorSubject<boolean>(false);
 
-    getUser() {
-        // Add your code here
-    }
+  public name$: Observable<string | null> = this.name$$.asObservable();
```

Comment:
Keep a record of the full User object instead of separate name and isAdmin role.